Add guide for separate content repo (#1476)
tinacms/tina.io: content/guides/tinacms/separate-content-repo/guide.md

---
title: Separate Content Repo
last_edited: '2023-01-24T10:00:00.000Z'
---

Tina supports sourcing content from a separate Git repo. With this approach you can continue to define your
Tina config in your website repo, but documents will be saved elsewhere.

## Why?

You might want to do this for a variety of reasons, such as:

- To decouple your website code from your content files so your commit history isn't a mix of content updates and code updates
- To avoid having to give Tina Cloud write access to your website's code (only give access to your content repo)

> ## Walkthrough

Reviewer: ```suggestion
## Introduction
```

This guide will start by having you clone the "website repo" locally. From there, you'll be creating the
"content repo" so you can start by cloning the project [here](https://github.com/tinacms/separate-website-repo):

```sh
git clone git@github.com:tinacms/separate-website-repo.git
```

The workflow requirements are:

- You must have _both_ the "content repo" and the "website repo" checked out onto your local machine.
- You must provide the location of your content repo in your Tina config (more about that below).
- When making changes, you'll need to ensure you've pushed _both_ repos to Github, with the "content repo"
  changes first

## Create the content repository

> Using a separate content repo works by defining your config in the repo you'll use for _editing_ with the CMS (which is this repo). We'll call that the "website repo".

Reviewer: I'm a bit unclear which one is "this repo"?

> To set up the "content repo", we'll start with a simple `.mdx` file:

Reviewer: This paragraph feels a tad weird because we're talking about the website repo, then switch to the instruction of setting up the website repo

Author: What do you suggest instead?

Reviewer: > Using a separate content repo works by defining your config in the repo you'll use for editing with the CMS (which is this repo). We'll call that the website repo.

^ I'd probably suggest moving that stuff out of the "Create the content repository" heading. Maybe there should be a separate "Creating the website repository" section?

```sh
mkdir -p ../demo-content-repo/content/pages && touch ../demo-content-repo/content/pages/home.mdx && echo "Hello" >> ../demo-content-repo/content/pages/home.mdx
```

## Local Development

> Install the project's dependencies:

Reviewer: Might want to specify this is done in the website repo

```sh
yarn install
```

Run the project locally:

```sh
yarn dev
```

## Visit the page in edit mode

Open [http://localhost:3000/admin/index.html](http://localhost:3000/admin/index.html)

From here you can add more fields to you content models in `.tina/config.js`. [Visit the docs](https://tina.io/docs/schema/) to learn more about content modeling.

## Deploying your content repo

> Since your content is in a separate repo, we'll want to use _that_ repo to connect to Tina Cloud.

Reviewer: ```suggestion
Next we'll want to create a Tina Cloud project from our "Content Repo"
```

Author: ```suggestion
Next we'll want to create a Tina Cloud project from our _content repo_
```

Initialize Git in your content repo and push it to Github. From there [connect to Tina Cloud](https://tina.io/docs/tina-cloud/dashboard/projects/)

Once that's done, [create a token](https://tina.io/docs/tina-cloud/dashboard/projects/#read-only-tokens) for the `main` branch, or use `*` to allow all branches.

> Switch back to this repo (your "website repo") and set up an `.env` file to use when connecting to Tina Cloud

Reviewer: ```suggestion
Switch back to your "website repo" and set up an `.env` file to use when connecting to Tina Cloud
```
